### PR TITLE
Add new metadata-cleanup module

### PR DESCRIPTION
**Pull request by JayGreeeen**
* Added kubernetes deployment options for new metadata cleanup module.
* Renamed references; `cleanup` -> `path-cleanup`, and `path-scheduler` -> `scheduler-apiary`.

### CODE REVIEW

File: k8s-path-cleanup.tf [hunk 1]
```diff
@@ -3,75 +3,75 @@
...
-            container_port = var.k8s_scheduler_port
+            name           = local.path_cleanup_name
+            container_port = var.k8s_path_cleanup_port
           }
 
           liveness_probe {
             http_get {
               path = "/actuator/health"
-              port = var.k8s_scheduler_port
+              port = local.path_cleanup_name
```
Comment (nvitucci) [inline k8s-path-cleanup.tf:62]: Shouldn't this be `var.k8s_scheduler_port`?

**Review by nvitucci — CHANGES_REQUESTED** [review]
It looks good. There is only one configuration line which I think is wrong.

File: k8s-path-cleanup.tf (hunk 1 above)
Comment (JayGreeeen) [reply k8s-path-cleanup.tf:62]: For some reason these changes arent being recognised properly - I edited the `k8s-cleanup.tf` file, renamed it to `k8s-path-cleanup.tf` and everywhere it said `cleanup` I changed it to `path-cleanup`. In the original file this line says `port = local.cleanup_name` https://github.com/ExpediaGroup/apiary-lifecycle/blob/master/k8s-cleanup.tf#L62 but you have a point, it doesnt look right does it?

Comment (nvitucci) [reply k8s-path-cleanup.tf:62]: Hmm, in this case it may be right, since it refers to the port by name.

Comment (nvitucci) [reply k8s-path-cleanup.tf:62]: Have you tried to access the endpoint? Does it behave correctly? If yes, the port is correct.

Comment (JayGreeeen) [reply k8s-path-cleanup.tf:62]: Update: we have discussed this and its okay as the port is being referenced by name, see another example here https://github.com/kubernetes-csi/livenessprobe

**Review by nvitucci — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -58,17 +65,17 @@ If the chosen `db_password_strategy` is `aws-secrets-manager`, this terraform mo
...
+| k8s\_scheduler\_apiary\_port | Internal port that the Beekeeper Scheduler Apiary service runs on. | `number` | `8080` | no |
 | message\_retention\_seconds | SQS message retention (s). | `string` | `"604800"` | no |
-| path\_scheduler\_docker\_image | Beekeeper path-scheduler image. | `string` | `"expediagroup/beekeeper-path-scheduler-apiary"` | no |
-| path\_scheduler\_docker\_image\_version | Beekeeper path-scheduler image version. | `string` | `"latest"` | no |
-| path\_scheduler\_ecs\_cpu | The amount of CPU used to allocate for the Beekeeper Path Scheduler Apiary ECS task.<br>Valid values: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-cpu-memory-error.html | `string` | `"2048"` | no |
-| path\_scheduler\_ecs\_memory | The amount of memory (in MiB) used to allocate for the Beekeeper Path Scheduler Apiary container.<br>Valid values: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-cpu-memory-error.html | `string` | `"4096"` | no |
+| scheduler\_apiary\_docker\_image | Beekeeper Scheduler Apiary image. | `string` | `"expediagroup/beekeeper-scheduler-apiary"` | no |
```
Comment (abhimanyugupta07) [inline README.md:75]: README is missing `metadata_cleanup_docker_image_version` and `metadata_cleanup_docker_image`

File: variables.tf [hunk 3]
```diff
@@ -271,6 +284,12 @@ variable "prometheus_enabled" {
   type        = string
 }
 
+variable "metastore_uri" {
+  description = "Metastore URI."
```
Comment (abhimanyugupta07) [inline variables.tf:288]:
```suggestion
  description = "URI of the metastore where tables to be cleaned-up are located. Required for Beekeeper Metadata Cleanup."
```

You can use `terraform-docs markdown .` command to generate your README section for all variables.

**Review by abhimanyugupta07 — CHANGES_REQUESTED** [review]
Please run `terraform fmt` once.

File: README.md [hunk 4]
```diff
@@ -24,10 +24,10 @@ If the chosen `db_password_strategy` is `aws-secrets-manager`, this terraform mo
 | apiary\_metastore\_listener\_arn | ARN of the Apiary Metastore Listener. | `string` | n/a | yes |
 | aws\_region | AWS region to use for resources. | `string` | n/a | yes |
 | beekeeper\_tags | A map of tags to apply to resources. | `map(string)` | n/a | yes |
-| cleanup\_docker\_image | Beekeeper cleanup docker image. | `string` | `"expediagroup/beekeeper-cleanup"` | no |
-| cleanup\_docker\_image\_version | Beekeeper cleanup docker image version. | `string` | `"latest"` | no |
-| cleanup\_ecs\_cpu | The amount of CPU used to allocate for the Beekeeper Cleanup ECS task.<br>Valid values: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-cpu-memory-error.html | `string` | `"2048"` | no |
-| cleanup\_ecs\_memory | The amount of memory (in MiB) used to allocate for the Beekeeper Cleanup container.<br>Valid values: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-cpu-memory-error.html | `string` | `"4096"` | no |
+| path\_cleanup\_docker\_image | Beekeeper Path Cleanup  docker image. | `string` | `"expediagroup/beekeeper-path-cleanup"` | no |
```
[the remark this answers is not in the document]
Comment (JayGreeeen) [reply README.md:27]: yeah

File: README.md (hunk 2 above)
Comment (JayGreeeen) [reply README.md:75]: nice spot!

**Review by abhimanyugupta07 — APPROVED** [review]

**Review by abhimanyugupta07 — APPROVED** [review]

**Review by vedantchokshi — APPROVED** [review]

**Pull request merged**